Makefile

```diff
@@ -1,14 +1,32 @@
-.PHONY: all test tests j2me java certs app clean jasmin aot shumway config-build
-BASIC_SRCS=$(shell find . -maxdepth 2 -name "*.ts" -not -path "./build/*")
+.PHONY: all test tests j2me java certs app clean jasmin aot shumway
+BASIC_SRCS=$(shell find . -maxdepth 2 -name "*.ts" -not -path "./build/*") config.ts
 JIT_SRCS=$(shell find jit -name "*.ts" -not -path "./build/*")
 SHUMWAY_SRCS=$(shell find shumway -name "*.ts")
 RELEASE ?= 0
+PROFILE ?= $(if $(findstring 0,$(RELEASE)),1,0)
 
-all: config-build java jasmin tests j2me shumway aot
+# Create a checksum file to monitor the changes of the Makefile configuration.
+# If the configuration has changed, we update the checksum file to let the files
+# which depend on it to regenerate.
+
+CHECKSUM := "$(RELEASE)$(PROFILE)"
+OLD_CHECKSUM := "$(shell [ -f .checksum ] && cat .checksum)"
+$(shell [ $(CHECKSUM) != $(OLD_CHECKSUM) ] && echo $(CHECKSUM) > .checksum)
```

Comment:
Ah, nice. I'd been wondering how we can do this. This is simple and elegant. Let's remember to update it when we add more build configuration variables.